warehouse/warehouse.go

```diff
@@ -514,22 +518,24 @@ func uploadFrequencyExceeded(warehouse warehouseutils.WarehouseT, syncFrequency
 	return false
 }
 
-func setLastProcessedMarker(warehouse warehouseutils.WarehouseT) {
+func setLastProcessedMarker(warehouse warehouseutils.WarehouseT, lastProcessedTime time.Time) {
 	lastProcessedMarkerMapLock.Lock()
 	defer lastProcessedMarkerMapLock.Unlock()
-	lastProcessedMarkerMap[warehouse.Identifier] = time.Now().Unix()
+	lastProcessedMarkerMap[warehouse.Identifier] = lastProcessedTime.Unix()
 }
 
-func (wh *HandleT) createUploadJobsFromStagingFiles(warehouse warehouseutils.WarehouseT, whManager manager.ManagerI, stagingFilesList []*StagingFileT, priority int) {
+func (wh *HandleT) createUploadJobsFromStagingFiles(warehouse warehouseutils.WarehouseT, whManager manager.ManagerI, stagingFilesList []*StagingFileT, priority int) time.Time {
 	// count := 0
 	// Process staging files in batches of stagingFilesBatchSize
 	// Eg. If there are 1000 pending staging files and stagingFilesBatchSize is 100,
 	// Then we create 10 new entries in wh_uploads table each with 100 staging files
 	var stagingFilesInUpload []*StagingFileT
 	var counter int
 	uploadTriggered := isUploadTriggered(warehouse)
+	uploadStartAfter := getUploadStartAfterTime()
```

Comment:
Why not pass it as an argument to the function ?
The reason I am saying it is because it makes the function definition ambiguous and hard to follow.